lib/shared/errors.js

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+/**
+ * @summary Human-friendly error messages
+ * @namespace HUMAN_FRIENDLY
+ * @public
+ */
+exports.HUMAN_FRIENDLY = {
+
+  /* eslint-disable new-cap */
+
+  /**
+   * @namespace ENOENT
+   * @memberof HUMAN_FRIENDLY
+   */
+  ENOENT: {
+
+    /**
+     * @property {Function} title
+     * @memberof ENOENT
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    title: (error) => {
+      return `No such file or directory: ${error.path}`;
+    },
+
+    /**
+     * @property {Function} description
+     * @memberof ENOENT
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    description: _.constant('The file you\'re trying to access doesn\'t exist')
```

Comment:
@jhermsmeier Some error messages depend on the error content. Check `ENOENT` a bit above for an example. The purpose of making them functions is that you can interpolate the error contents if needed, plus create if conditionals based on the environment, etc, which might be useful (we used to use that for the Resin CLI for example).

I guess we could make errors that don't need such feature strings, and check the type in the helper function. What do you think?